src/main/java/org/ohdsi/authenticator/service/GoogleIapJwtVerifier.java

```diff
@@ -0,0 +1,101 @@
+package org.ohdsi.authenticator.service;
+
+import com.google.common.base.Preconditions;
+import com.nimbusds.jose.JWSHeader;
+import com.nimbusds.jose.JWSVerifier;
+import com.nimbusds.jose.crypto.ECDSAVerifier;
+import com.nimbusds.jose.jwk.ECKey;
+import com.nimbusds.jose.jwk.JWK;
+import com.nimbusds.jose.jwk.JWKSet;
+import com.nimbusds.jwt.JWTClaimsSet;
+import com.nimbusds.jwt.SignedJWT;
+import java.net.URL;
+import java.security.interfaces.ECPublicKey;
+import java.time.Clock;
+import java.time.Instant;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+import org.ohdsi.authenticator.exception.AuthenticationException;
+import org.springframework.stereotype.Component;
+
+@Component
+public class GoogleIapJwtVerifier {
```

Comment:
Why do you need it to be a bean?

Reply:
Why not? I like DI approach, and I just delegated the responsibility of creating this class to the spring. This way It makes code easy for mocking/testing